lib/codecs/src/encoding/mod.rs

```diff
@@ -317,9 +331,10 @@ pub enum Serializer {
 }
 
 impl Serializer {
-    /// Check if the serializer supports encoding to JSON via `Serializer::to_json_value`.
-    pub fn supports_json(&self) -> bool {
+    /// Check if the serializer supports encoding an event to JSON via `Serializer::to_json_value`.
+    pub fn supports_json(&self, event: &Event) -> bool {
         match self {
+            Serializer::Gelf(_) => event.maybe_as_log().is_some(),
```

Comment:
I don't think this change is necessary.

The `GelfSerializer` already informed the sink that it only accepts `DataType::Log`, so this will always be true